app/src/main/java/com/m/helper/MobileService.java

```diff
@@ -0,0 +1,162 @@
+    public String getGraphqlServer() {
+        SyncApp service = (SyncApp) config.getServiceByType(SYNC_APP);
+        return service.getUrl();
+    }
+
+    @Override
+    public String getKIssuer() {
+        Keycloak service = (Keycloak) config.getServiceByType(KEYCLOAK);
+        String server = service.getConfig().getAuth_server_url();
+        String realm = service.getConfig().getRealm();
+        return server + "/realms/" + realm + "/";
+    }
+
+    @Override
+    public String getKClientId() {
+        Keycloak service = (Keycloak) config.getServiceByType(KEYCLOAK);
+        return service.getConfig().getResource();
+    }
+
+    @Override
+    public String getPushUrl() {
+        Push service = (Push) config.getServiceByType(PUSH);
+        return service.getUrl();
+    }
+
+    @Override
+    public String getPushVariantId() {
+        Push service = (Push) config.getServiceByType(PUSH);
+        return service.getConfig().getVariantId();
+    }
+
+    @Override
+    public String getPushVariantSecret() {
+        Push service = (Push) config.getServiceByType(PUSH);
+        return service.getConfig().getVariantSecret();
+    }
+
+    private void readMobileServiceJSON(){
+        try {
+            BufferedReader bufferedReader = getBufferReader();
+            JsonObject array = getJsonObject(bufferedReader);
+
+            config.setClientId(array.get("clientId").getAsString());
+            config.setNamespace(array.get("namespace").getAsString());
+
+            JsonArray services = array.get("services").getAsJsonArray();
+            configureServices(services);
+
+        } catch (FileNotFoundException e) {
+            System.out.println("APP: FileNotFoundException " + e.toString());
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.out.println("APP: IOException " + e.toString());
+
+        }
+    }
+
+    private Service buildService(JsonObject service){
+        Service result = null;
+        switch (service.get("type").getAsString()) {
+            case SYNC_APP:
+                result = gson.fromJson(service, SyncApp.class);
+                break;
+            case KEYCLOAK:
+                result = gson.fromJson(service, Keycloak.class);
+                break;
+            case PUSH:
+                Push r = gson.fromJson(service, Push.class);
+                JsonElement jsonConfig = service.get("config");
+                JsonElement jsonAndroidConfig = jsonConfig.getAsJsonObject().get("android");
+                AndroidConfig androidConfig = gson.fromJson(jsonAndroidConfig, AndroidConfig.class);
+                r.setConfig(androidConfig);
+                result = r;
+                break;
+            default:
+                System.out.println("APP: No type found");
+
+        }
+        return result;
+    }
+
+    private BufferedReader getBufferReader() throws IOException {
+        InputStream IS = assetManager.open("config/mobile-services.json");
```

Comment:
it's probably better to provide a way to allow user to configure the path of this file.

Reply:
Should I set this as a variable at the top of the class? I have found that the file has to be inside a folder that is nested under the `src/maim/assets` to be read.

Reply:
If you are using the `assetManager` then yes, the file needs to be in `src/main/assets` directory. It is ok to ask users to put the file in that directory, and provide a default recommended path, but it also should allow users to change the path.

Yes, you can set the variable at the top of the class.